### ISSUE TITLE
Test Code and Reproduce the results

### ISSUE THREAD

**Issue by tqosu** [Finspire13/DiffAct#1]
Hi Daochang,

Could you also update the inference instruction?

Based on the training script, I didn't obtain a result similar to what was reported in the paper.

For gtea dataset, results for 2 of 4 splits.

Epoch 10000 - decoder-agg-Test-Acc 79.82436882546652
Epoch 10000 - decoder-agg-Test-Edit 81.8243661100804
Epoch 10000 - decoder-agg-Test-F1@10 87.76978417266189
Epoch 10000 - decoder-agg-Test-F1@25 87.05035971223022
Epoch 10000 - decoder-agg-Test-F1@50 80.5755395683453

Epoch 10000 - decoder-agg-Test-Acc 78.19410319410319
Epoch 10000 - decoder-agg-Test-Edit 79.28571428571429
Epoch 10000 - decoder-agg-Test-F1@10 85.39325842696628
Epoch 10000 - decoder-agg-Test-F1@25 83.89513108614231
Epoch 10000 - decoder-agg-Test-F1@50 77.90262172284645

Any suggestions?

Thanks.

**Comment by Finspire13** [comment]
Hi tqosu, 

Thank you very much for your interest and the suggestion.

Following previous practice (https://github.com/ttlmh/Bridge-Prompt/issues/2), we need to select the best result from earlier epochs. We select the epoch according to the AVG metric. We select the same epoch across splits. Hope it helps and feel free to let me know for any questions.

**Comment by tqosu** [comment]
Thanks. Are you planning to share your code for automatically generating the optimal model following the training process and the inference instruction?

**Comment by Finspire13** [comment]
Hi tqosu,

I can try to if I find some time. I think it will not be very hard for you to implement it with a few lines of code, including loading the log files and some mean() and max() calls. Given a trained model, you can also use the test() function in main.py by specifying the 'model_path' to evaluate it.

**Comment by tqosu** [comment]
Thanks for the tips. Yes, I agree it is not hard. But since you previously used it in the paper, it would be easier for you to upload directly and also would make this repository a complete implementation.

**Comment by Finspire13** [comment]
Hi tqosu,

Here is one of the codes used before to summarize the results. You may need to make some minor changes to suit the current version of the code. I will update a clean version to the repo together with some instructions later. Thank you very much for the suggestion!

Alternatively, the models in the "trained_models" folder can be loaded to reproduce the results.

```
import os
import numpy as np

root_dir = './result'

def summary_results(prefix, opts, repeats, splits, epochs, mode, train_test='test'):

    all_results = np.zeros((len(opts), len(repeats), len(splits), len(epochs), 5))

    missing_count = 0
    for opt_id, opt in enumerate(opts):
        for repeat_id, repeat in enumerate(repeats):
            for split_id, split in enumerate(splits):

                if opt is None:
                    result_dir = os.path.join(root_dir, f'{prefix}-S{split}-{repeat}')
                else:
                    result_dir = os.path.join(root_dir, f'{prefix}-S{split}-{repeat}-{opt_id}')
                
                for epoch_id, epoch in enumerate(epochs):

                    result_file = os.path.join(result_dir, f'{train_test}_results_{mode}_epoch{epoch}.npy')

                    try:
                        result = np.load(result_file, allow_pickle=True).item()
                    except:
                        all_results[opt_id, repeat_id, split_id, epoch_id] = all_results[opt_id, repeat_id, split_id, epoch_id-1]
                        missing_count += 1
                        continue
                    
                    all_results[opt_id, repeat_id, split_id, epoch_id, 0] = result['Acc']
                    all_results[opt_id, repeat_id, split_id, epoch_id, 1] = result['Edit']
                    all_results[opt_id, repeat_id, split_id, epoch_id, 2] = result['F1@10']
                    all_results[opt_id, repeat_id, split_id, epoch_id, 3] = result['F1@25']
                    all_results[opt_id, repeat_id, split_id, epoch_id, 4] = result['F1@50']

    print(f'Missing: {missing_count}')
        
    return all_results


def get_best_epochs(results, epochs, window_size):
    # results: (epoch_num, )

    max_value = 0
    max_index = -1
    for o in range(len(epochs)-window_size+1):
        if results[o:o+window_size].mean() > max_value:
            max_value = results[o:o+window_size].mean()
            max_index = o
        
    return epochs[max_index:max_index+window_size]


prefix = '50salads-Reproduce37'

opts = [None]

splits = [1, 2, 3, 4, 5]
epochs = [i for i in range(0, 5001, 100)]
window_size = 1
mode = 'decoder-agg'
repeats = [0, 1, 2]

all_results = summary_results(prefix, opts, repeats, splits, epochs, mode, train_test='test')

print(all_results.mean(0).mean(0).mean(0).mean(-1).max())

for opt_id, opt in enumerate(opts):

    best_epochs = get_best_epochs(all_results[opt_id].mean(0).mean(0).mean(1), epochs, window_size)
    best_epoch_ids = [epochs.index(i) for i in best_epochs]
    # print(best_epoch_ids)

    results = all_results[opt_id].mean(0).mean(0)[best_epoch_ids].mean(0)
    
    print(best_epochs, results.mean())
    
    print('Opt', opt)
    print('-------------')
    print(results.mean())
    print('-------------')
    print(results)
    print('-------------')
    print(f'multicolumn{{3}}{{c}}{{ {results[2]:.1f} / {results[3]:.1f} / {results[4]:.1f} }} & {results[1]:.1f} & {results[0]:.1f} & {results.mean():.1f}')
    print('-------------')

    print()
    print()
```

**Comment by tqosu** [comment]
Great. Thanks!
